File: example-fs/Aoc.fs
open System
open Puzzle

let runner =
    match Environment.GetEnvironmentVariable("part") with
    | "part1" -> Ok(runPartOne)
    | "part2" -> Ok(runPartTwo)
    | part -> Error(part)

match runner with
| Ok(runPuzzle) ->
    printfn "Result: %s" (
        IO.File.ReadAllText(Environment.GetCommandLineArgs()[1]).TrimEnd()
        |> parseInput
        |> runPuzzle
        |> string
    )
| Error(part) -> printfn "Unknown part: '%s'" part

File: day05/aoc.js
import { parseInput, runPartOne, runPartTwo } from "./puzzle";

const runPuzzle = { part1: runPartOne, part2: runPartTwo }[process.env.part];
if (runPuzzle) {
  (async () => console.log("Result:", runPuzzle(parseInput((await Bun.file(process.argv[2]).text()).trimEnd()))))();
} else {
  console.log(`Unknown part: '${process.env.part ?? ""}'`);
}
